### PR TITLE
Add support for OS X.

### PR DESCRIPTION
**Pull request by Lukasa**
This is based on the work done for PyCA cryptography.

### CODE REVIEW

**Comment by SethMichaelLarson** [comment]
Never understood why still to this day it's impossible to test Python on Mac OS with Travis.

**Comment by Lukasa** [comment]
It's not impossible, but it requires much more manual work. Still, this isn't too bad.

**Comment by SethMichaelLarson** [comment]
Oh definitely not impossible, but not as easy as using Linux.

**Comment by Lukasa** [comment]
Sure, but just you wait until you want some Windows.

**Comment by SethMichaelLarson** [comment]
I've actually had good experiences with AppyVeyor :P

**Comment by Lukasa** [comment]
Sure, it's fine, but it's not *less* work. =D

**Comment by SethMichaelLarson** [comment]
Yeah it's really not haha, gotta write your own Python installer script if you want any versions besides 2.7.9 and 3.5 I believe.

**Comment by SethMichaelLarson** [comment]
Looks like PyPy is failing a single SSL test, so close!

**Comment by Lukasa** [comment]
Worse, it's failing a single SSL test *on OS X*. This is almost certainly because of `c_rehash`, which I *hate*.

**Comment by SethMichaelLarson** [comment]
Let the hate flow through you! :) You got this

**Comment by Lukasa** [comment]
Hrm, that doesn't seem to be it, the newest OpenSSL doesn't have a different rehash policy.

**Comment by SethMichaelLarson** [comment]
Is it something to do with `DEFAULT_CA_DIR`? I know the CA stuff is tougher to figure out in Mac OS/Windows. Sorry I can't be of more help in this.

**Comment by Lukasa** [comment]
Should be: the cert validation is *failing*, which is usually the result of failing to find the right cert, and having DEFAULT_CA_DIR set wouldn't lead to that.

**Comment by Lukasa** [comment]
But as a good first step, I can reproduce this failure locally. So: progress!

**Comment by SethMichaelLarson** [comment]
Huzzah to that! That's great news!

**Comment by Lukasa** [comment]
Ok, this error seems to be the result of PyPy being linked against the ancient OpenSSL in macOS. For whatever reason, that OpenSSL doesn't seem to parse the cadir properly. Given that I can reproduce the problem using the command-line version of openssl on my own mac, I'm just going to throw my hands in the air for now and try to avoid getting blocked behind that.

Let's remove the PyPy support for now. We can consider adding it again later.

File: _travis/install.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    pip install --user virtualenv
+else
+    pip install virtualenv
+fi
+
+pip install tox
+
+if [[ "${TOXENV}" == "gae" && ! -d ${GAE_PYTHONPATH} ]]; then
+  python _travis/fetch_gae_sdk.py ;
+fi
```
Comment (haikuginger) [inline _travis/install.sh]: No \n makes me sad. 😜

**Comment by SethMichaelLarson** [comment]
@Lukasa Okay, sounds like a good way to go for now.

File: _travis/install.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            ;;
+        py34)
+            pyenv install 3.4.5
+            pyenv global 3.4.5
+            ;;
+        py35)
+            pyenv install 3.5.2
+            pyenv global 3.5.2
+            ;;
+        pypy*)
```
Comment (haikuginger) [inline _travis/install.sh]: Remove this if we're not testing macOS with PyPy?

Comment (SethMichaelLarson) [inline _travis/install.sh]: We may in the future though, there's just an issue with PyPy's OpenSSL linking.

Comment (Lukasa) [inline _travis/install.sh]: Yeah, on balance I'm kinda inclined to leave it.

Comment (haikuginger) [inline _travis/install.sh]: 👍

**Comment by SethMichaelLarson** [comment]
Of course the build fails randomly right after passing. :P Can something be done about that Address already in use error? I've seen it a few times in automated builds randomly.

**Comment by Lukasa** [comment]
We covered it in most places, but it has clearly not totally gone away. These are usually timing issues.

**Comment by SethMichaelLarson** [comment]
@Lukasa Could you restart the py27 build now, then it'll be done before the Mac OS builds :P

**Comment by Lukasa** [comment]
Alright, cool, that's OS X. Next up: coverage reports.

**Pull request merged**